backend/src/model/dto.rs

```diff
@@ -63,10 +63,25 @@ pub struct PlantsSummaryDto {
     pub common_name: Option<Vec<Option<String>>>,
 }
 
-#[allow(clippy::missing_docs_in_private_items)] // TODO: See #97.
+/// An array of PlantsSummaryDto with some basic pagination information used by the plants search endpoint.
+#[typeshare]
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, ToSchema)]
+pub struct PlantsSearchDto {
+    /// A single page of searched plants.
+    pub plants: Vec<PlantsSummaryDto>,
+    /// Flag indicating whether there is more data on the next page.
+    pub has_more: bool,
+}
```

Comment:
We will have to solve this problem of not loading everything at once for other list endpoints as well.
I think we should generalize this into reusable offset-based pagination.
This would not only make things more uniform not only within this endpoint (response body is a list without params but an object with) but also for the whole API.

@markus2330 
I can create an issue and work on it in the coming week as my other work is probably blocked by the new plant columns anyway.

Reply:
> I can create an issue and work on it in the coming week as my other work is probably blocked by the new plant columns anyway.

Yes please. I would prefer if we did this in another PR to keep this one at a manageable size.

Reply:
Yes, great offer! :heart: